packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -185,40 +186,27 @@ const authMiddleware: AuthMiddleware = (...args: unknown[]) => {
     }
 
     // TODO: fix type discrepancy between WithAuthOptions and AuthenticateRequestOptions
-    const requestState = await authenticateRequest(req, options as AuthenticateRequestOptions);
-    const requestStateHeaders = requestState.headers;
-
-    const locationHeader = requestStateHeaders?.get('location');
-
-    // triggering a handshake redirect
-    if (locationHeader) {
-      return new Response(null, { status: 307, headers: requestStateHeaders });
-    }
+    const authenticateRequestOptions = {
+      ...options,
+      secretKey: options.secretKey || SECRET_KEY,
+      publishableKey: options.publishableKey || PUBLISHABLE_KEY,
+      ...handleMultiDomainAndProxy(req, options as AuthenticateRequestOptions),
+    } as AuthenticateRequestOptions;
+    const requestState = await clerkClient.authenticateRequest(req, authenticateRequestOptions);
+
+    if (requestState.status === AuthStatus.Handshake) {
```

Comment:
Got it, makes sense.
I think we might want to introduce a new status for it or at least think of a different way to transfer this knowledge into the code as it's hard to follow the logic here otherwise

Reply:
Check updated